src/main/java/mate/academy/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,93 @@
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@RequiredArgsConstructor
+@Service
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartRepository cartRepository;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository itemRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    @Transactional
+    public OrderDto createOrder(Long userId, CreateOrderRequestDto orderDto) {
+        ShoppingCart cart = cartRepository.findByUserId(userId);
+        if (cart.getCartItems().isEmpty()) {
+            throw new OrderException("Cart is empty for user: " + userId);
+        }
+        Order order = orderMapper.cartToOrder(cart, orderDto.shippingAddress());
+        cart.clearCart();
+        return orderMapper.toOrderDto(orderRepository.save(order));
+    }
+
+    @Override
+    public List<OrderDto> getAllOrders(Long userId, Pageable pageable) {
+        return orderMapper.toOrderDtoList(orderRepository.findAllByUserId(userId, pageable));
+    }
+
+    @Override
+    public UpdateOrderResponseDto updateOrderStatus(Long orderId, UpdateOrderDto orderDto) {
+        Order order = orderRepository.findById(orderId)
+                .map(o -> {
+                    o.setStatus(getStatusByName(orderDto.status()));
+                    return o;
+                })
+                .orElseThrow(() -> new EntityNotFoundException(
+                        String.format("Order with id: %d not found", orderId)
+                ));
+        return orderMapper.toUpdateDto(orderRepository.save(order));
+    }
+
+    @Override
+    public List<OrderItemDto> findOrderItemsByOrder(Long userId, Long orderId) {
+        Order order = orderRepository.findByIdAndUserId(orderId, userId)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        String.format("Order with id: %d not found for user: %d", orderId, userId)
+                ));
+        return orderItemMapper.toOrderItemDtoList(order.getOrderItems());
+    }
+
+    @Override
+    public OrderItemDto findOrderItemById(Long orderId, Long orderItemId, Long userId) {
+        OrderItem item = itemRepository.findByIdAndOrderIdAndUserId(orderItemId, orderId, userId)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        String.format("Order item: %d not found in order: %d for user: %d",
+                                orderItemId, orderId, userId)
+                ));
+        return orderItemMapper.toOrderItemDto(item);
+    }
+
+    private Status getStatusByName(String status) {
+        try {
+            return Status.valueOf(status.toUpperCase());
+        } catch (IllegalArgumentException e) {
+            throw new EntityNotFoundException(
+                    String.format("Status: %s not found", status), e
+            );
+        }
```

Comment:
Usually, the status is selected from the drop-down menu, so I don't think that this check is mandatory

Reply:
removed